### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by dhonsel**

### CODE REVIEW

File: common.mk [hunk 1]
```diff
@@ -12,3 +12,10 @@ ZIPINPUT = true
 # libgsl-dev, libboost-all-dev, libsuitesparse-dev, liblpsolve55-dev, libmysql++-dev and libsqlite3-dev 
 # are not available
 COMPGENEPRED = true
+
+# set TESTING to false if you don't need testing for the comparative gene prediction mode (CGP) or
+# the required libraries 
+# libboost-serialization-dev 
+# are not available
+TESTING = true
```
Comment (MarioStanke) [inline common.mk:20]: Hold on, until we have an answer from Giovanna, whether this default is really not a large risk in master (installation, runtime). Possibly we set this as developers only.

Comment (MarioStanke) [reply common.mk:20]: @mglgnn and I talked. 
Please move the TESTING = true to src/Makefile, as users are asked to edit common.mk but not src/Makefile.
After the pull is done and the testing branch merged, Giovanna will use the very test to confirm that some upcoming refactoring that regards TESTING is not braking anything in both settings TESTING=true/false.

Comment (dhonsel) [reply common.mk]: Moved TESTING to src/Makefile.

**Comment by dhonsel** [comment]
> * `README-testing.md` is in the main directory but contains stuff that is not ready or for everybody. Something about 'dramatically reduced from 1-3 Gigas to a few Megas', and  'random sampling of chunks'  which is internal documentation.  It is OK to have internal documentation, but it should be more hidden, like in docs, and clearly marked to be intended for developers of augustus.
>   The title of the file.
>   The command lines are for testing CGP (multi genome), but this is not expected from the file name or headings.
>   Move this file to `docs/longrunning-test-CGP.md`, we can later link from the main README.mk to some docs/testing.md which itself contains the whole picture or links to the different tests (short, long, for users, developers, performance).

The file is now moved.


> * data in examples/cgp12way
>   We had said we put the data online outside the repo. I can put a .tar.gz or similar on our web server as I did with my single-genome longrunning tests and it can be downloaded when the test is executed. git submodules for the test data could be an alternative. In both cases ,regular users would not download and store this data. And it would not be in the .deb package.

I would prefer the web server variant as for your single-genome longrunning test case. Access to the required data is easy for anyone who wants to run the test case.

**Comment by mglgnn** [comment]
For what regards documentation, no problem, the contents of README-testing.md was intended to be fixed. In the final version, I'll keep only details the end user is supposed to be interested in, as requested.

I've changed the code so that now we have:
1. if TESTING is false all snippets responsible for testing are not included
2. if TESTING is true, you can still decide either to run test or original prediction depending on how the cmd line parameter /Testing/testMode is set (**run** _vs_ **predict**, the latter being the default)
(we can also completely remove the command line parameter, but as a consequence the Python wrapper should be adjusted accordingly)

Code duplication has been removed. I've just to run the new version and check it does not break original prediction.

**Comment by dhonsel** [comment]
> Hi Daniel,
> can you please remove the data from this branch requested to be pulled into master?
> I will put them on our web server.
> It is ok if the longrunning tests fail in the short meantime.

Hi Mario,
the data in examples/cgp12way is now removed from branch testing.

**Comment by mglgnn** [comment]
I uploaded the latest version after removing code redundancy. Some refactoring is available which makes start within CompGenePred class much more readable, but we can integrate it later. maybe after discussing it.
So far I ran the following tests over the new code:
1. with TRAINING=true, I ran test for our data-set : ok
2. with TRAINING=true, I ran original prediction for our data-set : ok
3. with TRAINING=false, I ran original prediction for our data-set : ok
An extended test can be run to apply original prediction to all chromosome 1 and verify no side effects have been introduced (even if, after successful tests 1-2-3 this seems to be unlikely). It will take at least 24 hours.

I clean-up docs/longrunning-test-CGP.md leaving essential directions only.

**Comment by MarioStanke** [comment]
@mglgnn 
Giovanna, 

"uploaded the latest version"
to which branch? Any objections to merge testing into master now?

**Comment by mglgnn** [comment]
Branch testing_dataset, the one I created for beta version of testing.

Please, decide whether you prefer to wait for complete test over chromosome1 (only to verify no side effects occur for original prediction) or tests I reported are enough to greenlight merging of branch.

**Comment by mglgnn** [comment]
My opinion is that it is quite unlikely for any bug to have escaped tests so far run. I also checked the code several times. Further tests over chr1 are just for completeness

**Pull request merged**